src/sinks/util/rusoto2.rs

```diff
@@ -261,12 +246,16 @@ impl Body for RusotoBody {
                 }
             }
             Some(SignedRequestPayload::Stream(stream)) => {
-                match with_notify(cx, || stream.poll())? {
-                    Async::Ready(Some(buffer)) => {
-                        Poll::Ready(Some(Ok(io::Cursor::new(buffer.into_iter().collect()))))
-                    }
-                    Async::Ready(None) => Poll::Ready(None),
-                    Async::NotReady => Poll::Pending,
+                let stream = unsafe { Pin::new_unchecked(stream) };
```

Comment:
Not sure is it possible create pinned stream without `unsafe` here or not.... :thinking:

Reply:
Please include some reasoning why `Pin::new` is not appropriate and why this is safe. :)

Reply:
haha, I thought about this so much, though about `pin_project` or something, but `Pin::new` just works.
I hope that I'll get clear understanding of pin things at one day. Thanks Ana!